core/src/main/java/org/apache/metamodel/data/AbstractRowBuilder.java

```diff
@@ -68,7 +71,7 @@ public AbstractRowBuilder(Column[] columns) {
 
     @Override
     public final Row toRow() {
-        return new DefaultRow(new SimpleDataSetHeader(_columns), _values);
+        return new DefaultRow(new SimpleDataSetHeader(Arrays.stream(_columns).map(SelectItem::new).collect(Collectors.toList())), _values);
```

Comment:
Looks like it might be an advantage to make `_columns` a list here, instead. Of course that means a whole lot more changes in the class, so it certainly also makes sense not to.